File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {

    let right = height.length - 1, left = 0, currMaxAmount = 0;

    while (left < right) {
        let w = right - left;
        let h = Math.min(height[left], height[right]);
        let area = w * h;

        currMaxAmount = Math.max(area, currMaxAmount);

        if (height[left] > height[right])
            right--;
        else
            left++;
    }

    return currMaxAmount;
};

File: LeetCode/Medium/0011-container-with-most-water/0011-container-with-most-water.go
func maxArea(height []int) int {
	left := 0
	right := len(height) - 1
	currMaxArea := 0
	for left < right {
		temp := (right - left) * (min(height[left], height[right]))
		currMaxArea = max(currMaxArea, temp)
		if height[left] > height[right] {
			right--
		} else {
			left++
		}
	}
	return currMaxArea
}